dvc/dvcfile.py

```diff
@@ -225,9 +225,13 @@ def stage(self):
 
     @property
     def stages(self):
+        from dvc.parsing import DataLoader
+
         data, _ = self._load()
+        spec = DataLoader(data)
+        resolved = spec.resolve()
         lockfile_data = self._lockfile.load()
-        return StageLoader(self, data.get("stages", {}), lockfile_data)
+        return StageLoader(self, resolved.get("stages", {}), lockfile_data)
```

Comment:
Hack for now, we'll need to short-circuit this function to the repo somehow.

I like `repo.stages.get()` syntax (that means, we will need to think something for current `repo.stages`.